lib/types/string/index.js

```diff
@@ -351,45 +354,32 @@ internals.String = class extends Any {
                 let version = guidOptions.version[i];
                 Hoek.assert(typeof version === 'string', 'version at position ' + i + ' must be a string');
                 version = version.toLowerCase();
-                Hoek.assert(uuids[version], 'version at position ' + i + ' must be one of ' + Object.keys(uuids).join(', '));
-                Hoek.assert(versions.indexOf(version) === -1, 'version at position ' + i + ' must not be a duplicate.');
-                versions.push(version);
+                const versionNumber = uuids[version];
+                Hoek.assert(versionNumber, 'version at position ' + i + ' must be one of ' + Object.keys(uuids).join(', '));
+                Hoek.assert(!(versionNumber in versions), 'version at position ' + i + ' must not be a duplicate.');
+
+                versionNumbers.push(versionNumber);
+                versions[versionNumber] = '';
             }
+
+            versionNumbers = versionNumbers.join('');
```

Comment:
Yeah, I think I was just having a bout of temporary idiocy. Instead of `versionNumbers.push(versionNumber)` and `versionNumbers = versionNumbers.join('')` I could just replace line 361 with `versionNumbers += versionNumber` so that I am left with my concatenated string. 

The purpose of these parts is solely to be used for building out the RegExp.